homeassistant/components/sense/__init__.py

```diff
@@ -40,30 +40,12 @@
 type SenseConfigEntry = ConfigEntry[SenseData]
 
 
-class SenseDevicesData:
-    """Data for each sense device."""
-
-    def __init__(self):
-        """Create."""
-        self._data_by_device = {}
-
-    def set_devices_data(self, devices):
-        """Store a device update."""
-        self._data_by_device = {device["id"]: device for device in devices}
-
-    def get_device_by_id(self, sense_device_id):
-        """Get the latest device data."""
-        return self._data_by_device.get(sense_device_id)
-
-
 @dataclass(kw_only=True, slots=True)
 class SenseData:
     """Sense data type."""
 
     data: ASyncSenseable
-    device_data: SenseDevicesData
-    trends: DataUpdateCoordinator[None]
-    discovered: list[dict[str, Any]]
+    trends: DataUpdateCoordinator[Any]
```

Comment:
Why do we change to `Any`?

Reply:
mypy was generating an error if it's "None":

`homeassistant/components/sense/sensor.py:242: error: Argument 1 to "__init__" of "CoordinatorEntity" has incompatible type "DataUpdateCoordinator[None]"; expected "DataUpdateCoordinator[dict[str, Any]]"  [arg-type]
Found 1 error in 1 file (checked 2 source files)`

Reply:
You need to extend the coordinator entity with the coordinator type `CoordinatorEntity[DataUpdateCoordinator[None]]`